code/modules/projectiles/guns/projectile/constructable/siegecannon.dm

```diff
@@ -243,3 +274,69 @@
 		H.Knockdown(5)
 	else if(L.size == SIZE_TINY)	//splat
 		L.gib()
+
+//Bananium////////
+
+/obj/item/cannonball/bananium
+	name = "clownnonball"
+	desc = "A large sphere of honk."
+	icon = 'icons/obj/siege_cannon.dmi'
+	icon_state = "clownnonball"
+	starting_materials = list(MAT_CLOWN = CC_PER_SHEET_METAL*3)
+	adjRange = 50
+	adjSpeed = 1
+	adjForce = 0
+
+/obj/item/cannonball/bananium/throw_at(atom/target, range, speed, override = 1)
+	if(!cannonFired)
+		..()
+	else
+		..(target, 50, 1000, 1, 1)	//Always travels as slow as possible. High speed is to appease throw_at and doesn't translate to any damage
+
+/obj/item/cannonball/bananium/cannonEffect(var/atom/cTarg)
+	if(ishuman(cTarg))
+		honkMob(cTarg)
+	honkBounce(cTarg)
+
+/obj/item/cannonball/bananium/throw_impact(atom/hit_atom, var/speed, mob/user)
+	..()
+	if(isliving(hit_atom))
+		honkMob(hit_atom)
+
+/obj/item/cannonball/bananium/proc/honkMob(var/mob/living/L)
+	L.Knockdown(rand(2,10))
+	playsound(src, 'sound/items/bikehorn.ogg', 75, 1)
+	honkBounce(L)
+
+/obj/item/cannonball/bananium/proc/honkBounce(var/atom/cTarg)
+	var/list/honkStep = alldirs.Copy()
+	var/honkDir = get_dir(src, cTarg)
+	honkStep -= list(honkDir, turn(honkDir, 45), turn(honkDir, -45))	//Every direction possible except directly, or diagonally, toward what we hit
+	honkDir = pick(honkStep)
+	spawn(3)	//Prevents multiple instances of throw_at() from being active
+		bounceStep(honkDir)
+
+/obj/item/cannonball/bananium/proc/bounceStep(var/honkDir)
+	if(cannonFired)
+		if(prob(10) && istype(get_turf(src), /turf/simulated))
+			var/turf/simulated/T = get_turf(src)
+			T.wet(800, TURF_WET_LUBE)
+		var/target = get_ranged_target_turf(src, honkDir, 50)
+		throw_at(target, 50, 1000, 1, 1)
+
+/obj/item/cannonball/bananium/proc/stopBouncing()
+	throwing = 0
+	kinetic_acceleration = 0
+	if(cannonFired)
+		cannonAdjust()
+
+/obj/item/cannonball/bananium/Crossed(atom/movable/A)	//Yes, you have to arrest the cannonball
+	..()
+	if(istype(A, /obj/item/projectile))	//It's going to be moving a lot, might as well limit typechecking as much as possible
+		if(istype(A, /obj/item/projectile/energy/electrode) || istype(A, /obj/item/projectile/ricochet/taser))
```

Comment:
>you can stop it with the ricochet taser

very powerful

wish there was a way to not hardcode this but nothing springs to mind other than turning this into a mob itself

Reply:
I can add ```projectile/energy/proc/isTaser()``` if you'd like but it seems kind of wasteful with just these two types. 
And ya, the ricochet taser is easily the best way to stop it. Fire with fire and all that.